libs/blocks/basic-brick/basic-brick.js

```diff
@@ -0,0 +1,97 @@
+import { decorateTextOverrides, decorateBlockText, decorateBlockBg, decorateIconStack } from '../../utils/decorate.js';
+import { createTag } from '../../utils/utils.js';
+
+const blockTypeSizes = {
+  full: ['xxl', 'm', 'l'],
+  default: ['xl', 'm', 'l'],
+};
+const objFitOptions = ['fill', 'contain', 'cover', 'none', 'scale-down'];
+
+function getBlockSize(el) {
+  const sizes = Object.keys(blockTypeSizes);
+  const size = sizes.find((s) => el.classList.contains(`${s}-grid`)) || 'default';
+  return blockTypeSizes[size];
+}
+
+function handleBrickFragment(el) {
+  if (!el.closest('.fragment')) return;
+  const gridVariant = [...el.classList]?.find((c) => c.match(/-grid/));
+  if (!gridVariant || !el.closest('.section.masonry-layout')) return;
+  if (!el.parentNode.classList.contains('.section.masonry-layout')) {
+    el.closest('.section.masonry-layout div:not([class])')?.classList.add(gridVariant);
+  }
+}
+
+function decorateDefaultButton(foreground) {
+  if (!foreground.querySelector('.action-area')) return;
+  const btns = foreground.querySelectorAll('.con-button');
+  [...btns].forEach((btn) => btn.classList.add('button-l'));
+  const nextP = foreground.querySelector('.action-area + p');
+  const lastP = foreground.querySelector('.action-area ~ p:last-child');
+  if (nextP) nextP.className = '';
+  if (lastP) lastP.className = 'supplemental-text';
+}
+
+function handleObjectFit(bgRow) {
+  const bgConfig = bgRow.querySelectorAll('div');
+  [...bgConfig].forEach((r) => {
+    const image = r.querySelector('img');
+    if (!image) return;
+    const pchild = r.querySelectorAll(':scope > p');
+    if (pchild.length < 2) return;
```

Comment:
You may want to check `<p>` **_only_** if it has _**content**_ to avoid empty `<p>` case.

Reply:
@seanchoi-dev I have made the suggested change and am now targetting p:not(:empty) for further processing.